src/SigasiProjectCreator/ArgsAndFileParser.py

```diff
@@ -4,37 +4,40 @@
     :license: BSD, see LICENSE for more details.
 """
 import os
-from optparse import OptionParser
+import argparse
 
 
 class ArgsAndFileParser:
     options = None
 
     def __init__(self, usage):
-        self.parser = OptionParser(usage=usage)
-        self.parser.add_option('-l', '--layout', action="store", dest="layout",
-                               choices=['default', 'simulator'], default='default')
-
-    def parse_args(self, args_len, optional_dir=False):
-        (ArgsAndFileParser.options, args) = self.parser.parse_args()
-
-        if len(args) < args_len:
-            self.parser.error("incorrect number of arguments")
-        if optional_dir:
-            self.check_next_arg_is_dir(args, args_len)
+        self.parser = argparse.ArgumentParser(prog='SigasiProjectCreator')
+        self.parser.add_argument('project_name', help='project name')
+        self.parser.add_argument('input_file', help='input file')
+        self.parser.add_argument('destination_folder', help='root folder of created project')
+        self.parser.add_argument('-l', '--layout', action='store', dest='layout',
+                                 choices=['default', 'simulator'], default='default',
+                                 help='Project layout: default (in place) or simulator (one folder per library with linked files)')
+
+    def parse_args(self):
+        args = self.parser.parse_args()
+        if args.project_name is None:
+            self.parser.error("Incorrect number of arguments: project name missing")
+        if args.input_file is None:
+            self.parser.error("Incorrect number of arguments: input file missing")
+        if not os.path.isfile(args.input_file):
+            self.parser.error("Incorrect number of arguments: given input file does not exist")
+        if args.destination_folder is not None:
+            if not os.path.isdir(args.destination_folder):
+                self.parser.error("destination folder has to be a folder")
         return args
 
-    def check_next_arg_is_dir(self, args, args_len):
-        if len(args) > args_len and not os.path.isdir(args[args_len]):
-            self.parser.error("destination has to be a folder")
-
     def parse_args_and_file(self, parse_file):
-        args = self.parse_args(2, True)
-        project_name = args[0]
-        input_file = args[1]
-        destination = args[2] if len(args) > 2 else os.getcwd()
-        entries = parse_file(input_file)
-        return project_name, input_file, destination, entries
+        args = self.parse_args()
+        ArgsAndFileParser.options = args
```

Comment:
Why is this a class variable? (Not related to this patch)

Reply:
Because we want to access command line options from other places in the code, irrespective of a particular parser instance. It may not be the nicest approach, we can improve this if/when we do a larger overhaul of the project creator.